### ISSUE TITLE
[Feature]: (Willing to PR) Avoid KV cache occupying GPU memory when not used

### ISSUE THREAD

**Issue by fzyzcjy** [vllm-project/vllm#11408]
### 🚀 The feature, motivation and pitch

Hi thank you for the library! The use case is that, when doing online PPO, I hope to use vllm to generate llm completions, and then use RL to do gradient descent on those completions.

The problem is, to do this on a single GPU, the timeline is "vllm generate - Torch backward - repeat it". Thus, when torch doing backprop, I hope vllm can free its KV cache memory consumption, otherwise torch will not have enough memory.

Thanks for any suggestions!

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by houseroad** [comment]
Looks like the features are completed in https://github.com/vllm-project/vllm/issues/11408. Shall we close this one?
